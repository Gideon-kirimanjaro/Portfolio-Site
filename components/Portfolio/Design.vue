<template>
  <GridRow :data="data" />
</template>

<script>
import GridRow from '../UI/GridRow.vue'
export default {
  name: 'Design',
  components: {
    GridRow,
  },
  data() {
    return {
      data: [
        { key: 1, url: 1 },
        { key: 2, url: 2 },
        { key: 3, url: 3 },
        { key: 4, url: 4 },
        { key: 5, url: 5 },
        { key: 6, url: 6 },
        { key: 7, url: 7 },
        { key: 9, url: 9 },
        { key: 10, url: 10 },
        // { key: 11, url: 11 },
        { key: 12, url: 12 },
        { key: 13, url: 13 },
        { key: 14, url: 14 },
        { key: 15, url: 15 },
        { key: 16, url: 16 },
        { key: 17, url: 17 },
        { key: 18, url: 18 },
        { key: 19, url: 19 },
        { key: 21, url: 21 },
        { key: 22, url: 22 },
        { key: 23, url: 23 },
        { key: 24, url: 24 },
        { key: 25, url: 25 },
        { key: 26, url: 26 },
        { key: 27, url: 27 },
        { key: 28, url: 28 },
        { key: 29, url: 29 },
        // { key: 30, url: 30 },
        // { key: 31, url: 31 },
        { key: 32, url: 32 },
        { key: 33, url: 33 },
        { key: 34, url: 34 },
        { key: 35, url: 35 },
        { key: 36, url: 36 },
        { key: 37, url: 37 },
        { key: 38, url: 38 },
        { key: 39, url: 39 },
        { key: 40, url: 40 },
        { key: 41, url: 41 },
        { key: 42, url: 42 },
        { key: 43, url: 43 },
        { key: 44, url: 44 },
        { key: 45, url: 45 },
        { key: 46, url: 46 },
        { key: 47, url: 47 },
        { key: 48, url: 48 },
        { key: 49, url: 49 },
        { key: 50, url: 50 },
      ],
    }
  },
  methods: {
    keyGen() {
      for (let i = 0; i < 51; i++) {
        i = i + 1
      }
    },
  },
}
</script>

<style lang="scss" scoped></style>
