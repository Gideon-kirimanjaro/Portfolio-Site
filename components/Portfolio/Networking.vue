<template>
  <div id="network" class="container">
    <h3 class="d-flex justify-content-center text-warning font-weight-bold">
      <u>Networking & Support</u>
    </h3>
    <div
      v-for="(image, i) in netData"
      :key="i"
      class="container d-md-flex py-4"
    >
      <div id="imgDiv" class="">
        <img
          class="mt-2"
          :src="require(`@/static/images/${image.name}`)"
          alt=""
        />
      </div>
      <div class="mt-md-5">
        <a id="header-link" class="px-md-5 text-warning" target="_blank">
          {{ image.header }}
        </a>
        <p id="description" class="px-md-5 text-light">
          {{ image.description }}.
        </p>
        <div class="px-md-5">
          <nuxt-link target="_blank" to="/support">
            <b-button variant="success">View it here</b-button></nuxt-link
          >
        </div>
        <hr
          style="width: 100%; text-align: left; margin-right: 0 height:2px;border;color:gray;background-color:yellow"
        />
      </div>
    </div>

    <div class="d-flex justify-content-center">
      <NuxtLink target="_blank" to="/support">
        <b-button variant="success" class="mb-2">Show more</b-button></NuxtLink
      >
    </div>
  </div>
</template>

<script>
export default {
  name: 'Networking',
  components: {},
  computed: {
    netData() {
      const data = this.$store.state.networking.netData
      const sortedData = data.slice(0, 3)
      return sortedData
    },
  },
}
</script>

<style scoped>
* {
  font-family: montserrat;
}
#network {
  font-family: montserrat;
  background-color: blueviolet;
}

#header-link {
  color: rgb(116, 168, 11);
  font-size: 23px;
  font-weight: bolder;
}
#description {
  font-family: montserrat;
  font-size: 1.2rem;
}
@font-face {
  font-family: montserrat;
  src: url('@/static/fonts/Montserrat-Regular.ttf');
}
</style>
