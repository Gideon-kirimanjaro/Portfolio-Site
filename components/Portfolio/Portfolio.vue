<template>
  <div class="portfolio">
    <h2 class="d-flex justify-content-center">
      <u> PORTFOLIO </u>
    </h2>
    <div class="d-flex justify-content-center p-3">
      <img height="200" width="200" src="@/static/images/man.png" alt="" />
    </div>
    <h4
      class="d-flex justify-content-center text-success mt-2 mb-4 px-5 font-weight-bold"
    >
      A glimpse of the projects I've been working on.
    </h4>
    <div>
      <b-tabs pills card>
        <b-tab title="Web" active
          ><b-card-text class="d-flex justify-content-center"
            ><Web /></b-card-text
        ></b-tab>
        <b-tab title="Graphic Design"
          ><b-card-text class="d-flex justify-content-center"
            ><Design /></b-card-text
        ></b-tab>
        <b-tab title="Support"
          ><b-card-text class="d-flex justify-content-center"
            ><Networking /></b-card-text
        ></b-tab>
        <b-tab title="Data"
          ><b-card-text class="d-flex justify-content-center"
            ><Dash /></b-card-text
        ></b-tab>
      </b-tabs>
    </div>
    <!-- <div class="port2">
      <div ref="web">
        <Web />
      </div>
      <div ref="dash">
        <Dash />
      </div>
      <div ref="networking">
        <Networking />
      </div>
    </div> -->
  </div>
</template>

<script>
import Dash from '../Portfolio/Dash'
import Web from '../Portfolio/Web'
import Networking from '../Portfolio/Networking'
import Design from './Design.vue'
export default {
  name: 'Portfolio',
  components: {
    Dash,
    Web,
    Networking,
    Design,
  },
  methods: {
    // showData(refName) {
    //   const element = this.$refs[refName]
    //   const top = element.offsetTop
    //   window.scrollTo(0, top)
    // },
    // showWeb(refName) {
    //   const element = this.$refs[refName]
    //   const top = element.offsetTop
    //   window.scrollTo(0, top)
    // },
  },
}
</script>

<style scoped>
* {
  font-family: montserrat;
  src: url('@/static/fonts/Montserrat-Medium.ttf');
  box-sizing: border-box;
  margin: 0%;
  padding: 0%;
}
.portfolio {
  background-color: #ffffff;
  padding-top: 20px;
}
.port2 {
  margin-left: 20px;
}
</style>
