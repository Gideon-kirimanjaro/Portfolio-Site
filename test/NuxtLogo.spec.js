import { shallowMount } from '@vue/test-utils'
import Portfolio from '../components/Portfolio/Portfolio'

describe('Portfolio', () => {
  test('NuxtLogo renders properly', () => {
    const wrapper = shallowMount(Portfolio, {})
    expect(wrapper.html()).toMatchSnapshot()
  })
})
