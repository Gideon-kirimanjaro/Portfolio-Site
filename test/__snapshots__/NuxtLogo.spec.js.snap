// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio NuxtLogo renders properly 1`] = `
"<div class=\\"portfolio\\">
  <h2 class=\\"d-flex justify-content-center\\"><u> PORTFOLIO </u></h2>
  <div class=\\"d-flex justify-content-center p-3\\"><img height=\\"200\\" width=\\"200\\" src=\\"@/static/images/man.png\\" alt=\\"\\"></div>
  <h4 class=\\"d-flex justify-content-center text-success mt-2 mb-4 px-5 font-weight-bold\\">
    A glimpse of the projects I've been working on.
  </h4>
  <div>
    <b-tabs pills=\\"\\" card=\\"\\">
      <b-tab title=\\"Web\\" active=\\"\\">
        <b-card-text class=\\"d-flex justify-content-center\\">
          <web-stub></web-stub>
        </b-card-text>
      </b-tab>
      <b-tab title=\\"Graphic Design\\">
        <b-card-text class=\\"d-flex justify-content-center\\">
          <design-stub></design-stub>
        </b-card-text>
      </b-tab>
      <b-tab title=\\"Support\\">
        <b-card-text class=\\"d-flex justify-content-center\\">
          <networking-stub></networking-stub>
        </b-card-text>
      </b-tab>
      <b-tab title=\\"Data\\">
        <b-card-text class=\\"d-flex justify-content-center\\">
          <dash-stub></dash-stub>
        </b-card-text>
      </b-tab>
    </b-tabs>
  </div>
</div>"
`;
